samples/solidity/contracts/ERC721WithData.sol

```diff
@@ -112,14 +105,13 @@ contract ERC721WithData is Context, Ownable, ERC721, IERC721WithData {
     }
 
     function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
-        require(_exists(tokenId), 'ERC721WithData: Token does not exist');
-
+        _requireOwned(tokenId);
         string memory uri = _tokenURIs[tokenId];
         return uri;
     }
 
     function _setTokenURI(uint256 tokenId, string memory _tokenURI) internal {
```

Comment:
I didn't do it here, but I wonder if we should be inheriting https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721URIStorage.sol